src/storm/storage/jani/Model.cpp

```diff
@@ -1100,52 +1100,52 @@ Model& Model::replaceUnassignedVariablesWithConstants() {
     return *this;
 }
 
-Model& Model::substituteConstantsInPlace() {
+Model& Model::substituteConstantsInPlace(bool const& substituteTranscendentalNumbers) {
     // Gather all defining expressions of constants.
     std::map<storm::expressions::Variable, storm::expressions::Expression> constantSubstitution;
     for (auto& constant : this->getConstants()) {
         if (constant.hasConstraint()) {
-            constant.setConstraintExpression(substituteJaniExpression(constant.getConstraintExpression(), constantSubstitution));
+            constant.setConstraintExpression(
+                substituteJaniExpression(constant.getConstraintExpression(), constantSubstitution, substituteTranscendentalNumbers));
         }
         if (constant.isDefined()) {
-            constant.define(substituteJaniExpression(constant.getExpression(), constantSubstitution));
+            constant.define(substituteJaniExpression(constant.getExpression(), constantSubstitution, substituteTranscendentalNumbers));
             constantSubstitution[constant.getExpressionVariable()] = constant.getExpression();
         }
     }
 
     for (auto& functionDefinition : this->getGlobalFunctionDefinitions()) {
-        functionDefinition.second.substitute(constantSubstitution);
+        functionDefinition.second.substitute(constantSubstitution, substituteTranscendentalNumbers);
     }
 
     // Substitute constants in all global variables.
-    this->getGlobalVariables().substitute(constantSubstitution);
+    this->getGlobalVariables().substitute(constantSubstitution, substituteTranscendentalNumbers);
 
     // Substitute constants in initial states expression.
-    this->setInitialStatesRestriction(substituteJaniExpression(this->getInitialStatesRestriction(), constantSubstitution));
+    this->setInitialStatesRestriction(substituteJaniExpression(this->getInitialStatesRestriction(), constantSubstitution, substituteTranscendentalNumbers));
 
     for (auto& rewMod : this->getNonTrivialRewardExpressions()) {
-        rewMod.second = substituteJaniExpression(rewMod.second, constantSubstitution);
+        rewMod.second = substituteJaniExpression(rewMod.second, constantSubstitution, substituteTranscendentalNumbers);
     }
 
     // Substitute constants in variables of automata and their edges.
     for (auto& automaton : this->getAutomata()) {
-        automaton.substitute(constantSubstitution);
+        automaton.substitute(constantSubstitution, substituteTranscendentalNumbers);
     }
-
     return *this;
 }
 
 Model Model::substituteConstants() const {
     Model result(*this);
     result.replaceUnassignedVariablesWithConstants();
-    result.substituteConstantsInPlace();
+    result.substituteConstantsInPlace(false);
     return result;
 }
 
 Model Model::substituteConstantsFunctions() const {
```

Comment:
I added to this method the substitution of Transcendental numbers with Rational numbers, too.
Do you think it makes sense to reflect that in the name of the method, too? Something like `substituteConstantsFunctionsTranscendentals`?